lib/s3.persist.stream.js

```diff
@@ -0,0 +1,127 @@
+var AWS = require('aws-sdk');
+var _ = require('underscore');
+
+var stream = require('stream');
+var util = require('util');
+
+var rotateHistoryCount = 3;
```

Comment:
I agree. I actually have my own reservations on who should be responsible for the rotation.